xkcd.py

```diff
@@ -1,11 +1,56 @@
-from flask import Flask
+from flask import Flask, request, jsonify, abort
+import urllib.request, json
+from datetime import datetime
+import os
+
 app = Flask(__name__)
 
-@app.route('/comics/current')
+PORT =  5000 if os.environ.get('FLASK_PORT') is None else os.environ.get('FLASK_PORT')
+HOST = '127.0.0.1' if os.environ.get('FLASK_HOST') is None else os.environ.get('FLASK_Host')
+
+@app.route('/comics/current', methods=['GET'])
 def current_comic():
-    pass
+    current_comic_url = "http://xkcd.com/info.0.json"
+    return comic_by_url(current_comic_url)
+
+
+@app.route('/comics/<int:comic_id>', methods=['GET'])
+def comic_by_id(comic_id):
+    return comic_by_url(create_url_from_id(comic_id)), 200
+
+ 
+@app.route('/comics/many', methods=['GET'])
+def many_comics():
+    # we are assuming we don't want repetition and don't care about order of comics
+    comics_ids = set(request.args.to_dict(flat=False).get("comic_ids"))
+    return jsonify( [comic_by_url(create_url_from_id(comic_id)) for comic_id in comics_ids] )
+       
+
+@app.route('/comics/<string:_>')
+def wrong_url(_):
+    abort(400)
+
+
+def create_url_from_id(comic_id):
+    return f"http://xkcd.com/{comic_id}/info.0.json"
```

Comment:
there's one edge case which isn't right:

```
http "localhost:5000/comics/many?comic_ids=1&comic_ids=2000A"                                                 ✔  19:28:17 
HTTP/1.0 404 NOT FOUND
Content-Length: 232
Content-Type: text/html; charset=utf-8
Date: Tue, 24 Nov 2020 18:30:48 GMT
Server: Werkzeug/1.0.1 Python/3.8.5

<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2 Final//EN">
<title>404 Not Found</title>
<h1>Not Found</h1>
<p>The requested URL was not found on the server. If you entered the URL manually please check your spelling and try again.</p>
```

When I ask for one proper id and one invalid, I get `404`. A `400` should be returned (because one of the ids is not an integer)